Sources/MongoSwift/ReadConcern.swift

```diff
@@ -14,85 +14,104 @@ public class ReadConcern: Codable {
         case linearizable
         /// See https://docs.mongodb.com/master/reference/read-concern-snapshot/
         case snapshot
-    }
-
-    /// A pointer to a `mongoc_read_concern_t`.
-    internal var _readConcern: OpaquePointer?
+        /// Unknown level
+        case other(level: String)
+
+        public var rawValue: String {
+            switch self {
+            case .local:
+                return "local"
+            case .available:
+                return "available"
+            case .majority:
+                return "majority"
+            case .linearizable:
+                return "linearizable"
+            case .snapshot:
+                return "snapshot"
+            case .other(let l):
+                return l
+            }
+        }
 
-    /// The level of this `ReadConcern`, or `nil` if the level is not set.
-    public var level: String? {
-        guard let level = mongoc_read_concern_get_level(self._readConcern) else {
-            return nil
+        public init?(rawValue: String) {
+            switch rawValue {
+            case "local":
+                self = .local
+            case "available":
+                self = .available
+            case "majority":
+                self = .majority
+            case "linearizable":
+                self = .linearizable
+            default:
+                self = .other(level: rawValue)
+            }
         }
-        return String(cString: level)
     }
 
+    /// The level of this `ReadConcern`, or `nil` if the level is not set.
+    public var level: Level?
+
     /// Indicates whether this `ReadConcern` is the server default.
     public var isDefault: Bool {
-        return mongoc_read_concern_is_default(self._readConcern)
+       return level == nil
+    }
+
+    // Initializes a new `ReadConcern` with the same level as the provided `mongoc_read_concern_t`.
+    // The caller is responsible for freeing the original `mongoc_read_concern_t`.
+    internal init(from readConcern: OpaquePointer) {
+        if let level = mongoc_read_concern_get_level(readConcern) {
+            self.level = Level(rawValue: String(cString: level))
+        }
     }
 
-    /// Initialize a new `ReadConcern` from a `ReadConcern.Level`.
-    public convenience init(_ level: Level) {
-        self.init(level.rawValue)
+    /// Initializes a new `ReadConcern` by copying an existing `ReadConcern`.
+    public init(from readConcern: ReadConcern) {
```

Comment:
do we use this initializer anywhere?
now that ReadConcern is a value type, I'm not sure how useful it really is, since if you need a copy you can just do `let copyOfRC = oldRC`. (if you do that with a class, both variables still point to the same object.)